### PR TITLE
[gui] Kill g_audioManager

### PR DESCRIPTION
**Pull request by hudokkow**
## Motivation and Context
Kill a global following @FernetMenta's https://github.com/xbmc/xbmc/pull/13724 example.

## How Has This Been Tested?
Runtime tested on Ubuntu and Win10 x64.

### CODE REVIEW

File: xbmc/interfaces/json-rpc/InputOperations.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ JSONRPC_STATUS CInputOperations::SendAction(int actionID, bool wakeScreensaver /
   if(!wakeScreensaver || !handleScreenSaver())
   {
     g_application.ResetSystemIdleTimer();
-    g_audioManager.PlayActionSound(actionID);
+    CServiceBroker::GetGUI()->GetAudioManager().PlayActionSound(actionID);
```
Comment (Rechi) [inline xbmc/interfaces/json-rpc/InputOperations.cpp]:
```
    CGUIComponent* gui = CServiceBroker::GetGUI();
    if (gui)
      gui->GetAudioManager().PlayActionSound(actionID);
```

File: xbmc/addons/UISoundsResource.cpp [hunk 2]
```diff
@@ -42,7 +42,7 @@ bool CUISoundsResource::IsInUse() const
 void CUISoundsResource::OnPostInstall(bool update, bool modal)
 {
   if (IsInUse())
-    g_audioManager.Load();
+    CServiceBroker::GetGUI()->GetAudioManager().Load();
```
Comment (FernetMenta) [inline xbmc/addons/UISoundsResource.cpp]: this needs a check for nullptr. here we can't assume that there is a GUI

File: xbmc/input/InputManager.cpp [hunk 3]
```diff
@@ -697,11 +697,11 @@ bool CInputManager::ExecuteInputAction(const CAction &action)
   {
     bResult = g_application.OnAction(action);
     if (bResult)
-      g_audioManager.PlayActionSound(action);
+      CServiceBroker::GetGUI()->GetAudioManager().PlayActionSound(action);
```
Comment (FernetMenta) [inline xbmc/input/InputManager.cpp]: check for nullptr

File: xbmc/input/InputManager.cpp [hunk 4]
```diff
@@ -697,11 +697,11 @@ bool CInputManager::ExecuteInputAction(const CAction &action)
...
   }
   else
   {
-    g_audioManager.PlayActionSound(action);
+    CServiceBroker::GetGUI()->GetAudioManager().PlayActionSound(action);
```
Comment (FernetMenta) [inline xbmc/input/InputManager.cpp]: same here

**Comment by FernetMenta** [comment]
all calls from non gui components should test for nullptr

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by hudokkow** [comment]
Done. Thanks for the review.

File: xbmc/Application.cpp [hunk 5]
```diff
@@ -2096,7 +2096,7 @@ bool CApplication::OnAction(const CAction &action)
       if (!m_appPlayer.IsPaused() && m_appPlayer.GetPlaySpeed() != 1)
         m_appPlayer.SetPlaySpeed(1);
 
-      g_audioManager.Enable(m_appPlayer.IsPaused());
+      CServiceBroker::GetGUI()->GetAudioManager().Enable(m_appPlayer.IsPaused());
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: OnAction can be called without UI

File: xbmc/Application.cpp [hunk 6]
```diff
@@ -2156,7 +2158,7 @@ bool CApplication::OnAction(const CAction &action)
       {
         // unpause, and set the playspeed back to normal
         m_appPlayer.Pause();
-        g_audioManager.Enable(m_appPlayer.IsPaused());
+        CServiceBroker::GetGUI()->GetAudioManager().Enable(m_appPlayer.IsPaused());
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: needs nullptr check

File: xbmc/Application.cpp [hunk 7]
```diff
@@ -2869,7 +2871,7 @@ void CApplication::Stop(int exitCode)
     UnregisterActionListener(&m_appPlayer.GetSeekHandler());
     UnregisterActionListener(&CPlayerController::GetInstance());
 
-    g_audioManager.DeInitialize();
+    CServiceBroker::GetGUI()->GetAudioManager().DeInitialize();
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: needs nullptr check

File: xbmc/Application.cpp [hunk 8]
```diff
@@ -3205,7 +3207,7 @@ bool CApplication::PlayFile(CFileItem item, const std::string& player, bool bRes
   m_appPlayer.SetMute(m_muted);
 
 #if !defined(TARGET_POSIX)
-  g_audioManager.Enable(false);
+  CServiceBroker::GetGUI()->GetAudioManager().Enable(false);
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: needs nullptr check

File: xbmc/Application.cpp [hunk 9]
```diff
@@ -3218,7 +3220,7 @@ void CApplication::PlaybackCleanup()
 {
   if (!m_appPlayer.IsPlaying())
   {
-    g_audioManager.Enable(true);
+    CServiceBroker::GetGUI()->GetAudioManager().Enable(true);
```
Comment (FernetMenta) [inline xbmc/Application.cpp]: needs nullptr check

**Comment by FernetMenta** [comment]
finally player will not depend on GUI. player can also be used for transcoding

**Comment by hudokkow** [comment]
Done. Thanks for the guidance! :+1:
Transcoding sounds nice... :wink:

**Review by FernetMenta — APPROVED** [review]

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Pull request merged**